torch/testing/_internal/common_methods_invocations.py

```diff
@@ -146,6 +147,33 @@ def formatter(arg):
 
         return self._repr_helper(formatter)
 
+    def splat(self):
+        # TODO: Is this required? op(sample.splat()) doesn't work
+        # TODO: This needs comment
+        return (self.input, self.args, self.kwargs)
+
+    def numpy(self):
+        # Returns the NumPy version of the input
+        # Only supports torch.Tensor objects, if an arg is a scalar - a TypeError will be raised
+        sample_torch_input, sample_args, sample_kwargs = self.splat()
+
+        numpy_args = list()
+
+        for sample_arg in sample_args:
+            if isinstance(sample_arg, torch.Tensor):
+                numpy_arg = sample_arg.cpu().numpy()
+            else:
+                raise TypeError(f"Only torch.Tensor types supported but got {type(sample_arg)}")
+            numpy_args.append(numpy_arg)
+
+        if isinstance(sample_torch_input, torch.Tensor):
+            sample_numpy_input = sample_torch_input.cpu().numpy()
+        elif is_iterable_of_tensors(sample_torch_input):
+            sample_numpy_input = [sample_input.cpu().numpy() for sample_input in sample_torch_input]
+        else:
+            raise TypeError(f"Only torch.Tensor types support but got {type(sample_torch_input)}")
```

Comment:
For functions like `vstack`, the input is list of tensors and thus this case has been handled here. I don't think there will be a case when the input will be a scalar but an else loop won't hurt here?

Reply:
Just convert tensors to NumPy and preserve scalars/bools/strings etc. Handling containers in v0 is fine but be sure to call the conversion function recursively on containers. For now it could check for Sequences or Mappings.